src/main/java/io/supertokens/webserver/api/oauth/OAuthClientsAPI.java

```diff
@@ -63,25 +59,25 @@ public OAuthClientsAPI(Main main){
 
     @Override
     protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
-        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
 
-        boolean containsAllRequired = containsAllRequiredFields(input);
-        boolean containsMoreThanAllowed = containsMoreThanAllowed(input);
-        if(!containsAllRequired || containsMoreThanAllowed){
-            throw new ServletException(new WebserverAPI.BadRequestException("Invalid Json Input"));
-        }
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+        InputParser.collectAllMissingRequiredFieldsAndThrowError(input, REQUIRED_INPUT_FIELDS_FOR_POST);
 
         try {
             AppIdentifier appIdentifier = getAppIdentifier(req);
             Storage storage = enforcePublicTenantAndGetPublicTenantStorage(req);
 
             JsonObject response = OAuth.registerOAuthClient(super.main, appIdentifier, storage, input);
-            sendJsonResponse(200, response, resp);
+            JsonObject postResponseBody = new JsonObject();
+            postResponseBody.addProperty("status", "OK");
+            postResponseBody.add("client", response);
+            sendJsonResponse(200, postResponseBody, resp);
 
-        } catch (OAuthClientRegisterException registerException) {
+        } catch (OAuthClientRegisterInvalidInputException registerException) {
 
             JsonObject errorResponse = createJsonFromException(registerException);
-            sendJsonResponse(400, errorResponse, resp);
+            errorResponse.addProperty("status", "INVALID_INPUT");
+            sendJsonResponse(200, errorResponse, resp);
```

Comment:
this can happen cause of a programming error, so this should return a 400 to the client. Please also correct the comment in the GH issue

Reply:
corrected. Also, added the `_ERROR` postfix